src/test/checkqueue_tests.cpp

```diff
@@ -334,6 +334,8 @@ BOOST_AUTO_TEST_CASE(test_CheckQueue_FrozenCleanup)
         vChecks[0].should_freeze = true;
         control.Add(vChecks);
         control.Wait(); // Hangs here
+        bool waitResult = control.Wait(); // Hangs here
+        assert(waitResult);
```

Comment:
duplicate control.Wait() not sure what is correct here

Reply:
Ouch, good catch - I guess I've accepted both instead of accepting the new lines. Will update.